### PR TITLE
Job proposals can now be reproposed even when a proposal is already  approved

### PR DESCRIPTION
**Pull request by jkongie**
The new proposal behaviour is:

* A matching job proposal which exists as pending, should be updated to the newly proposed values
* A matching job proposal which exists as approved, should create a new job proposal as pending

### CODE REVIEW

File: core/services/feeds/service.go [hunk 1]
```diff
@@ -286,14 +286,13 @@ func (s *service) CreateJobProposal(jp *JobProposal) (int64, error) {
...
+// ProposeJob creates a job proposal if it does not exist. If a pending proposal
+// already exists, update the existing pending job proposal.
 //
-// The feeds manager id check exists for support of multiple feeds managers in
-// the future so that in the (very slim) off chance that the same uuid is
-// generated by another feeds manager or they maliciously send an existing uuid
-// belonging to another feeds manager, we do not update it.
+// We check that the job proposal belongs to the feeds manager who is making
+// the proposal because in anticipation for multiple feeds manager support. On
+// the highly unlikely event of a UUID collision, or a potential malicious
```
Comment (samsondav) [inline core/services/feeds/service.go:294]: UUID collision can be assumed to be impossible.

File: core/services/feeds/service.go [hunk 2]
```diff
@@ -286,14 +286,13 @@ func (s *service) CreateJobProposal(jp *JobProposal) (int64, error) {
...
+// attack by another feeds manager, we return an error.
```
Comment (samsondav) [inline core/services/feeds/service.go:295]: How would this attack look? Can you be specific?

**Review by samsondav — CHANGES_REQUESTED** [review]
Seems to be missing a test on the upsert/insert new distinction

File: core/services/feeds/service.go (hunk 2 above)
Comment (jkongie) [reply core/services/feeds/service.go:295]: If `feeds manager B` gets the remote UUID of job of `feeds manager A` they could craft a proposal with the same UUID, and change the existing proposal

**Comment by samsondav** [comment]
Could this be better implemented by introducing the idea of a version number?

**Comment by cl-homu** [comment]
:umbrella: The latest upstream changes (presumably 9ae2b4aafa22d60bde18337d502893f36748605a) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

**Comment by jkongie** [comment]
No longer valid. We now support versioning in https://github.com/smartcontractkit/chainlink/pull/5844

**Pull request closed without merging**